test/index.ts

```diff
@@ -51,5 +73,65 @@ describe('Mutation', () => {
       expect(+responseData.id).to.equal(storedUserData!.id); // responseData.id is a string
       expect(await compare(userInput.password, storedUserData!.password)).to.be.true;
     });
+
+    it('should give an error in case of short passwords', async () => {
+      const userInput = makeInput({ password: 'short' });
+      const responseData = (await requestUserCreation(userInput)).data.errors[0];
+      expect(responseData).excluding('stacktrace').to.deep.equal({
+        message: 'invalid password',
+        code: 400,
+        details: 'password should have at least 6 characters',
+      });
+      expect(await getNumberOfUsersInDB()).to.equal(0);
+    });
+
+    it('should give an error in case of passwords with no digits', async () => {
+      const userInput = makeInput({ password: 'abcdefghijklmnop' });
+      const responseData = (await requestUserCreation(userInput)).data.errors[0];
+      expect(responseData).excluding('stacktrace').to.deep.equal({
+        message: 'invalid password',
+        code: 400,
+        details: 'password should contain a digit',
+      });
+      expect(await getNumberOfUsersInDB()).to.equal(0);
+    });
+
+    it('should give an error in case of passwords with no letters', async () => {
+      const userInput = makeInput({ password: '123456789' });
+      const responseData = (await requestUserCreation(userInput)).data.errors[0];
+      expect(responseData).excluding('stacktrace').to.deep.equal({
+        message: 'invalid password',
+        code: 400,
+        details: 'password should contain a letter',
+      });
+      expect(await getNumberOfUsersInDB()).to.equal(0);
+    });
+
+    it('should give an error in case of emails with incorrect formats', async () => {
+      const userInput = makeInput({ email: 'bademail.com' });
+      const responseData = (await requestUserCreation(userInput)).data.errors[0];
+      expect(responseData).excluding('stacktrace').to.deep.equal({
+        message: 'invalid email',
+        code: 400,
+        details: 'email has invalid format',
+      });
+      expect(await getNumberOfUsersInDB()).to.equal(0);
+    });
+
+    it('should give an error in case of duplicated emails', async () => {
+      const existingUser = new User();
+      Object.assign(existingUser, defaultUserInput);
+      userRepository.save(existingUser);
```

Comment:
Fiquei com uma dúvida aqui: como esses testes são end-to-end seria mais correto eu criar o usuário mandando um request com o axios? Isso seria mais próximo do funcionamento real do servidor (que é a ideia dos testes e2e, pelo o que eu entendi), mas iria contra a independência dos testes, já que envolveria a mutation `createUser`, que é responsabilidade de outro caso de teste.